src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -927,102 +970,61 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
         /// </summary>
         /// <param name="resource">The <see cref="ODataResource"/> describing the resource, which is being annotated.</param>
         /// <param name="resourceContext">The context for the resource instance, which is being annotated.</param>        
-        public virtual void AppendInstanceAnnotations(ODataResource resource, ResourceContext resourceContext)
+        public virtual void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext)
         {
             IEdmStructuredType structuredType = resourceContext.StructuredType;
             IEdmStructuredObject structuredObject = resourceContext.EdmObject;
+
+            //For appending transient and persistent instance annotations for both enity object and normal resources
+            //var entityObject = structuredObject;
+
             PropertyInfo instanceAnnotationInfo = EdmLibHelpers.GetInstanceAnnotationsContainer(structuredType,
                 resourceContext.EdmModel);
 
-            object value;
-            
-            if (instanceAnnotationInfo == null || structuredObject == null ||
-                !structuredObject.TryGetPropertyValue(instanceAnnotationInfo.Name, out value) || value == null)
-            {
-                return;
-            }
+            EdmEntityObject edmEntityObject = null;
+            object value = null;
```

Comment:
>value [](http://example.com/codeflow?start=19&length=5)

can we use something more meaningful than "value" -- i.e., "instanceAnnotations"?